### PR TITLE
DAOS-8936 tests: Support detecting engine start via dmg

### PR DESCRIPTION
**Pull request by phender**
Enable tests to optionally detect if all the daos server engines have
started successfully via dmg system query instead of searching the
command output for engine started messages.

Quick-functional: true
Test-tag: daos_admin

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
Any conclusive results?

**Comment by phender** [comment]
Manual testing passed with latest changes.  Running through CI now.

File: src/tests/ftest/util/server_utils.py [hunk 1]
```diff
@@ -378,6 +389,55 @@ def detect_engine_start(self, host_qty=None):
...
         self._expected_states = self.get_current_state()
 
+    def get_detected_engine_count(self, sub_process):
+        """Get the number of detected joined engines.
+
+        Args:
+            sub_process (process.SubProcess): subprocess used to run the command
+
+        Returns:
+            int: number of patterns detected in the job output
```
Comment (shimizukko) [inline src/tests/ftest/util/server_utils.py:399]: It looks like this method returns bool.

**Review by shimizukko — APPROVED** [review]

**Review by mjean308 — APPROVED** [review]

File: src/tests/ftest/util/server_utils.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/util/server_utils.py:399]: Yes it does.  I'll fix it in a future PR.

**Comment by sylviachanoiyee** [comment]
I think we need to run more tests in addition to the daos admin priv test since this PR made changes to some common code used widely by other tests.
@phender , could you kick off another run without tag?
The queue is pretty empty and master landing is closed today due to CI issue so I think it's a good time to rerun. Thanks!

**Comment by phender** [comment]
The https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7425/5 re-run encountered 1 (unrelated) error:

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7425/5/artifact/Functional%20Hardware%20Medium/osa/offline_drain.py where the ior execution was unresponsive for 29 minutes before the 1830 second timeout was exceeded:

```
2021-11-29 22:01:32,744 process          L0604 INFO | Running '/usr/lib64/mpich/bin/mpirun -hostfile /var/tmp/avocado_9wd__8d2/avocado_job_b6726_8f/1-._osa_o_run-aggregation-akeys-single-checksum-container-dkeys-hosts-ior-clientslots-iorflags-mdtest-client_processes-wr_size-32K-pool-pool_capacity-rebuild-record-1KB-server_config-servers-0-1-server_manager-setup-snapshot-test_obj_class-test_ranks-65df/hostfile37470 -np 30 ior -a DFS -b 500000000 -w -F -k -G 1 -i 1 -o /testfile -t 500000 --dfs.chunk_size 1048576 --dfs.cont 88C69A3F-6B91-4933-9D5B-D3E139A73AD6 --dfs.dir_oclass RP_2G1 --dfs.group daos_server --dfs.oclass RP_2G1 --dfs.pool 54EDCBD5-9DCD-494F-A187-C55693B84CC0'
2021-11-29 22:01:32,747 process          L0677 INFO | Command /usr/lib64/mpich/bin/mpirun -hostfile /var/tmp/avocado_9wd__8d2/avocado_job_b6726_8f/1-._osa_o_run-aggregation-akeys-single-checksum-container-dkeys-hosts-ior-clientslots-iorflags-mdtest-client_processes-wr_size-32K-pool-pool_capacity-rebuild-record-1KB-server_config-servers-0-1-server_manager-setup-snapshot-test_obj_class-test_ranks-65df/hostfile37470 -np 30 ior -a DFS -b 500000000 -w -F -k -G 1 -i 1 -o /testfile -t 500000 --dfs.chunk_size 1048576 --dfs.cont 88C69A3F-6B91-4933-9D5B-D3E139A73AD6 --dfs.dir_oclass RP_2G1 --dfs.group daos_server --dfs.oclass RP_2G1 --dfs.pool 54EDCBD5-9DCD-494F-A187-C55693B84CC0 running on a thread
2021-11-29 22:01:33,588 process          L0416 DEBUG| [stdout] IOR-3.4.0+dev: MPI Coordinated Test of Parallel I/O
2021-11-29 22:01:33,589 process          L0416 DEBUG| [stdout] Began               : Mon Nov 29 22:01:33 2021
2021-11-29 22:01:33,589 process          L0416 DEBUG| [stdout] Command line        : ior -a DFS -b 500000000 -w -F -k -G 1 -i 1 -o /testfile -t 500000 --dfs.chunk_size 1048576 --dfs.cont 88C69A3F-6B91-4933-9D5B-D3E139A73AD6 --dfs.dir_oclass RP_2G1 --dfs.group daos_server --dfs.oclass RP_2G1 --dfs.pool 54EDCBD5-9DCD-494F-A187-C55693B84CC0
2021-11-29 22:01:33,589 process          L0416 DEBUG| [stdout] Machine             : Linux wolf-125.wolf.hpdd.intel.com
2021-11-29 22:01:35,484 process          L0416 DEBUG| [stdout] TestID              : 0
2021-11-29 22:01:35,484 process          L0416 DEBUG| [stdout] StartTime           : Mon Nov 29 22:01:35 2021
2021-11-29 22:01:35,485 process          L0416 DEBUG| [stdout] Path                : /testfile.00000000
2021-11-29 22:01:35,485 process          L0416 DEBUG| [stdout] FS                  : 670.6 GiB   Used FS: 14.6%   Inodes: -0.0 Mi   Used Inodes: 0.0%
2021-11-29 22:01:35,485 process          L0416 DEBUG| [stdout] 
2021-11-29 22:01:35,485 process          L0416 DEBUG| [stdout] Options: 
2021-11-29 22:01:35,485 process          L0416 DEBUG| [stdout] api                 : DFS
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] apiVersion          : DAOS
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] test filename       : /testfile
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] access              : file-per-process
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] type                : independent
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] segments            : 1
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] ordering in a file  : sequential
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] ordering inter file : no tasks offsets
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] nodes               : 1
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] tasks               : 30
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] clients per node    : 30
2021-11-29 22:01:35,486 process          L0416 DEBUG| [stdout] repetitions         : 1
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] xfersize            : 500000 bytes
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] blocksize           : 476.84 MiB
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] aggregate filesize  : 13.97 GiB
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] 
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] Results: 
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] 
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] access    bw(MiB/s)  IOPS       Latency(s)  block(KiB) xfer(KiB)  open(s)    wr/rd(s)   close(s)   total(s)   iter
2021-11-29 22:01:35,487 process          L0416 DEBUG| [stdout] ------    ---------  ----       ----------  ---------- ---------  --------   --------   --------   --------   ----
2021-11-29 22:30:47,633 stacktrace       L0039 ERROR| 
```

**Pull request merged**